Android Crash

fluttercontactpicker: ^3.1.0
crashed my whole app. :(
```
java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=2029, result=-1, data=Intent { dat=content://com.android.contacts/contacts/lookup/378i47ad7ed089a02a75.3789r3406-2A2E522A44/4149 flg=0x1 }} to activity {com.falsesilver.app/com.falsesilver.app.MainActivity}: java.lang.NullPointerException: it.getString(it.getColum…mmonDataKinds.Note.NOTE)) must not be null
E/AndroidRuntime(18548): 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4905)
E/AndroidRuntime(18548): 	at android.app.ActivityThread.handleSendResult(ActivityThread.java:4946)
E/AndroidRuntime(18548): 	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51)
E/AndroidRuntime(18548): 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
E/AndroidRuntime(18548): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
E/AndroidRuntime(18548): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2040)
E/AndroidRuntime(18548): 	at android.os.Handler.dispatchMessage(Handler.java:107)
E/AndroidRuntime(18548): 	at android.os.Looper.loop(Looper.java:224)
E/AndroidRuntime(18548): 	at android.app.ActivityThread.main(ActivityThread.java:7520)
E/AndroidRuntime(18548): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(18548): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
E/AndroidRuntime(18548): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
E/AndroidRuntime(18548): Caused by: java.lang.NullPointerException: it.getString(it.getColum…mmonDataKinds.Note.NOTE)) must not be null
E/AndroidRuntime(18548): 	at me.schlaubi.fluttercontactpicker.ContactPicker.getNote(ContactPicker.kt:112)
E/AndroidRuntime(18548): 	at me.schlaubi.fluttercontactpicker.ContactPicker.buildContact(ContactPicker.kt:99)
E/AndroidRuntime(18548): 	at me.schlaubi.fluttercontactpicker.ContactPicker.access$buildContact(ContactPicker.kt:18)
E/AndroidRuntime(18548): 	at me.schlaubi.fluttercontactpicker.ContactPicker$onActivityResult$3.invoke(ContactPicker.kt:47)
E/AndroidRuntime(18548): 	at me.schlaubi.fluttercontactpicker.ContactPicker$onActivityResult$3.invoke(ContactPicker.kt:18)
E/AndroidRuntime(18548): 	at me.schlaubi.fluttercontactpicker.ContactPicker.processContact(ContactPicker.kt:71)
E/AndroidRuntime(18548): 	at me.schlaubi.fluttercontactpicker.ContactPicker.onActivityResult(ContactPicker.kt:47)
E/AndroidRuntime(18548): 	at me.schlaubi.fluttercontactpicker.AbstractPickContext.onActivityResult(AbstractPickContext.kt:15)
E/AndroidRuntime(18548): 	at io.flutter.embedding.engine.FlutterEnginePluginRegistry$FlutterEngineActivityPluginBinding.onActivityResult(FlutterEnginePluginRegistry.java:691)
E/AndroidRuntime(18548): 	at io.flutter.embedding.engine.FlutterEnginePluginRegistry.onActivityResult(FlutterEnginePluginRegistry.java:378)
E/AndroidRuntime(18548): 	at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onActivityResult(FlutterActivityAndFragmentDelegate.java:625)
E/AndroidRuntime(18548): 	at io.flutter.embedding.android.FlutterActivity.onActivityResult(FlutterActivity.java:583)
E/AndroidRuntime(18548): 	at android.app.Activity.dispatchActivityResult(Activity.java:8250)
E/AndroidRuntime(18548): 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4898)
E/AndroidRuntime(18548): 	... 11 more


```

--- a/README.md
+++ b/README.md
@@ -1,5 +1,5 @@
 # fluttercontactpicker
-[![pub.dev](https://img.shields.io/badge/pub-3.1.0-green.svg)](https://pub.dev/packages/fluttercontactpicker#-readme-tab-)
+[![pub.dev](https://img.shields.io/badge/pub-3.1.1-green.svg)](https://pub.dev/packages/fluttercontactpicker#-readme-tab-)
 
 Interact with native OS contact pickers using Flutter
 

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -1,8 +1,8 @@
 group 'me.schlaubi.fluttercontactpicker'
-version '3.1.0-SNAPSHOT'
+version '3.1.1-SNAPSHOT'
 
 buildscript {
-    ext.kotlin_version = '1.4.10'
+    ext.kotlin_version = '1.4.20'
     repositories {
         google()
         jcenter()

--- a/android/src/main/kotlin/me/schlaubi/fluttercontactpicker/ContactPicker.kt
+++ b/android/src/main/kotlin/me/schlaubi/fluttercontactpicker/ContactPicker.kt
@@ -18,7 +18,7 @@ import java.io.ByteArrayOutputStream
 class ContactPicker private constructor(private val pickContext: PickContext, private val result: MethodChannel.Result, askForPermission: Boolean, private val requestCode: Int, private val type: Uri) : PluginRegistry.ActivityResultListener, PluginRegistry.RequestPermissionsResultListener {
 
     init {
-        val hasPermission = PermissionUtil.hasPermission(pickContext.context) || ("xaiomi" !in Build.MANUFACTURER /* Cool android OEMs think it's cool to not do what Android does */ && Build.VERSION.SDK_INT <= Build.VERSION_CODES.Q && requestCode != FlutterContactPickerPlugin.PICK_CONTACT) // below android 11 there is no need for permissions when only requesting email/phone number
+        val hasPermission = PermissionUtil.hasPermission(pickContext.context) || ("xiaomi" !in Build.MANUFACTURER.toLowerCase() /* Cool android OEMs think it's cool to not do what Android does */ && Build.VERSION.SDK_INT <= Build.VERSION_CODES.Q && requestCode != FlutterContactPickerPlugin.PICK_CONTACT) // below android 11 there is no need for permissions when only requesting email/phone number
         if (!hasPermission && askForPermission) {
             PermissionUtil.requestPermission(pickContext.activity, this)
         } else if (hasPermission) {

--- a/android/src/main/kotlin/me/schlaubi/fluttercontactpicker/ContactPicker.kt
+++ b/android/src/main/kotlin/me/schlaubi/fluttercontactpicker/ContactPicker.kt
@@ -18,7 +18,7 @@ import java.io.ByteArrayOutputStream
 class ContactPicker private constructor(private val pickContext: PickContext, private val result: MethodChannel.Result, askForPermission: Boolean, private val requestCode: Int, private val type: Uri) : PluginRegistry.ActivityResultListener, PluginRegistry.RequestPermissionsResultListener {
 
     init {
-        val hasPermission = PermissionUtil.hasPermission(pickContext.context) || (Build.VERSION.SDK_INT <= Build.VERSION_CODES.Q && requestCode != FlutterContactPickerPlugin.PICK_CONTACT) // below android 11 there is no need for permissions when only requesting email/phone number
+        val hasPermission = PermissionUtil.hasPermission(pickContext.context) || ("xaiomi" !in Build.MANUFACTURER /* Cool android OEMs think it's cool to not do what Android does */ && Build.VERSION.SDK_INT <= Build.VERSION_CODES.Q && requestCode != FlutterContactPickerPlugin.PICK_CONTACT) // below android 11 there is no need for permissions when only requesting email/phone number
         if (!hasPermission && askForPermission) {
             PermissionUtil.requestPermission(pickContext.activity, this)
         } else if (hasPermission) {

--- a/example/android/app/src/main/kotlin/com/example/example/MainActivity.kt
+++ b/example/android/app/src/main/kotlin/com/example/example/MainActivity.kt
@@ -2,4 +2,4 @@ package com.example.example
 
 import io.flutter.embedding.android.FlutterActivity
 
-class MainActivity: FlutterActivity()
+class MainActivity : FlutterActivity()

--- a/example/android/build.gradle
+++ b/example/android/build.gradle
@@ -1,12 +1,12 @@
 buildscript {
-    ext.kotlin_version = '1.4.10'
+    ext.kotlin_version = '1.4.20'
     repositories {
         google()
         jcenter()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.0.2'
+        classpath 'com.android.tools.build:gradle:4.1.1'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
     }
 }

--- a/example/android/gradle/wrapper/gradle-wrapper.properties
+++ b/example/android/gradle/wrapper/gradle-wrapper.properties
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.6.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-all.zip

--- a/ios/fluttercontactpicker.podspec
+++ b/ios/fluttercontactpicker.podspec
@@ -4,7 +4,7 @@
 #
 Pod::Spec.new do |s|
   s.name             = 'fluttercontactpicker'
-  s.version          = '3.1.0'
+  s.version          = '3.1.1'
   s.summary          = 'Interact with native OS contact pickers using Flutter'
   s.description      = <<-DESC
 Interact with native OS contact pickers using Flutter

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -1,6 +1,6 @@
 name: fluttercontactpicker
 description: Use Androids' and iOS' native contact pickers to chose phone numbers and email addresses from contacts of the users address book
-version: 3.1.0
+version: 3.1.1
 homepage: https://github.com/DRSchlaubi/contact_picker
 
 environment:

